src/main/java/org/folio/dcb/listener/kafka/CirculationEventListener.java

```diff
@@ -40,19 +40,20 @@ public class CirculationEventListener {
   public void handleLoanEvent(String data, MessageHeaders messageHeaders) {
     String tenantId = getHeaderValue(messageHeaders, XOkapiHeaders.TENANT, null).get(0);
     var eventData = parseLoanEvent(data);
-    if (Objects.nonNull(eventData)) {
+    if (Objects.nonNull(eventData) && eventData.isDcb()) {
```

Comment:
This condition should be changed and the case when isDcb flag is not present in  the payload the if condition should also pass. Please check